toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -243,6 +246,45 @@ func (b *LiveOSIsoBuilder) prepareRootfsForDracut(writeableRootfsDir string) err
 	return nil
 }
 
+// canonalizeExtraCommandLine
+//
+// given the kernel extra commandline parameters, this function ensures
+// that they are well-formatted for grub correct consumption.
+//
+// inputs:
+//   - extraCommandLine:
+//     extra commandline parameters to canonalize.
+//
+// outputs
+//   - canonalized commadline parameters.
+func canonalizeExtraCommandLine(extraCommandLine string) string {
+	cannonicalExtraCommandLine := ""
+	// Split by spaces/tabs into 'fields'.
+	fields := strings.Fields(extraCommandLine)
+	for _, field := range fields {
+		// If a field starts with 'ds=nocloud;s=' (for cloud-init), it means
+		// the user did not wrap it in single quotes as it should be. If left
+		// as-is, the grub parsing at runtime will drop the text after the ';'.
+		//
+		// Unfortunately, if the user adds single quotes, the higher-level
+		// toolkit and MIC schema validation fail because the Mariner Toolkit
+		// does not allow it.
+		//
+		// This Toolkit limitation is because it uses 'sed' to manipulate
```

Comment:
The escape character used with `sed` is the back-tick not single-quotes.

Also, the use of `sed` is causing problems then feel free to replace it with Go's regexp library. (Doing this is goodness regardless as we should avoid dependencies on external tools as much as possible.)

Reply:
updated the logic.